api/functions/routeDirectionsAPI.js

```diff
@@ -0,0 +1,102 @@
+import axios from "axios";
+export async function computeRoutesForOrigins(
+	meetingTimeArray,
+	arrayOfOrigins,
+	travelMode,
+	destinationParam,
+	fields,
+	apiKey,
+) {
+	const userTravelInfoContainer = [];
+
+	for (const arrivalTime of meetingTimeArray) {
+		const arrivalDetail = {
+			arrivalTime: arrivalTime,
+			details: [],
+		};
+
+		for (const originElement of arrayOfOrigins) {
+			try {
+				const requestBody = {
+					origin: {
+						location: {
+							latLng: originElement.location.latLng,
+						},
+					},
+					destination: {
+						location: {
+							latLng: {
+								latitude: destinationParam.latitude,
+								longitude: destinationParam.longitude,
+							},
+						},
+					},
+					travelMode,
+					arrivalTime, // Use the current arrival time from the loop
+				};
+
+				// Send request to Google Routes API
+				const response = await axios.post(
+					"https://routes.googleapis.com/directions/v2:computeRoutes",
+					requestBody,
+					{
+						headers: {
+							"Content-Type": "application/json",
+							"X-Goog-Api-Key": apiKey,
+							"X-Goog-FieldMask": fields,
+						},
+					},
+				);
+
+				// Ensure routes exist before accessing them
+				if (response.data.routes && response.data.routes.length > 0) {
+					const route = response.data.routes[0];
+					let firstDepartureTime = null;
+					let lastArrivalTime = null;
+					let duration = null;
+					let staticDuration = null;
+
+					for (const leg of route.legs) {
+						duration = leg.duration || duration;
+						staticDuration = leg.staticDuration || staticDuration;
+
+						for (const step of leg.steps) {
+							if (step.transitDetails) {
+								if (!firstDepartureTime) {
+									firstDepartureTime =
```

Comment:
I needed to capture the first departure time and the last arrival time. So, by looping through each step, when I get the first departure time, I keep it as is and don’t update it. But for the last arrival time, since some steps don’t have this value, I go through all the steps and store the last one in lastArrivalTime. Do I need to change this? I’m not sure which part is inconsistent.